src/odb/include/odb/dbWireCodec.h

```diff
@@ -631,6 +644,11 @@ class dbWireDecoder
   /// junction-id of the previous point from which this branch emerges.
   ///
   int getJunctionValue() const;
+
+  ///
+  /// Get the current mask color.
+  ///
+  std::optional<uint8_t> getColor() const;
```

Comment:
Does this swig correctly?

Reply:
```c++
SWIGINTERN int
_wrap_dbWireDecoder_getColor(ClientData clientData SWIGUNUSED, Tcl_Interp *interp, int objc, Tcl_Obj *CONST objv[]) {
  odb::dbWireDecoder *arg1 = (odb::dbWireDecoder *) 0 ;
  void *argp1 = 0 ;
  int res1 = 0 ;
  std::optional< uint8_t > result;
  
  if (SWIG_GetArgs(interp, objc, objv,"o:odb::dbWireDecoder_getColor self ",(void *)0) == TCL_ERROR) SWIG_fail;
  res1 = SWIG_ConvertPtr(objv[1], &argp1,SWIGTYPE_p_odb__dbWireDecoder, 0 |  0 );
  if (!SWIG_IsOK(res1)) {
    SWIG_exception_fail(SWIG_ArgError(res1), "in method '" "dbWireDecoder_getColor" "', argument " "1"" of type '" "odb::dbWireDecoder const *""'"); 
  }
  arg1 = reinterpret_cast< odb::dbWireDecoder * >(argp1);
  {
    try {
      result = ((odb::dbWireDecoder const *)arg1)->getColor(); 
    }
    catch (std::bad_alloc &) {
      fprintf(stderr, "Error: out of memory.");
      exit(0);
    }
    // This catches std::runtime_error (utl::error) and sta::Exception.
    catch (std::exception &excp) {
      Tcl_ResetResult(interp);
      Tcl_AppendResult(interp, excp.what(), nullptr);
      return TCL_ERROR;
    }
  }
  Tcl_SetObjResult(interp,SWIG_NewInstanceObj((new std::optional< uint8_t >(result)), SWIGTYPE_p_std__optionalT_uint8_t_t, SWIG_POINTER_OWN | 0 ));
  return TCL_OK;
fail:
  return TCL_ERROR;
}
```

It looks like it's doing something reasonable?

Reply:
Please make a unit test that demonstrates it.

Reply:
My guess is SWIGTYPE_p_std__optionalT_uint8_t_t is going to be an opaque type.

Reply:
Done.